guide/airline_reservation/airline_reservation_service.bal

```diff
@@ -0,0 +1,268 @@
+            done;
+        }
+
+        // Query the database to retrieve flight details
+        json flightDetails = untaint airlineDBService(airline, departureDate, arrivalDate, to, rom);
+        // Response payload
+        log:printDebug("Client response from Qatar : " + flightDetails.toString());
+        response.setJsonPayload(flightDetails);
+        // Send the response to the caller
+        _ = caller->respond(response);
+    }
+
+    // Resource 'flightAsiana', which checks about airline 'Asiana'
+    @http:ResourceConfig {
+        methods:["POST"],
+        path:"/asiana",
+        consumes:["application/json"],
+        produces:["application/json"]
+    }
+    flightAsiana (endpoint caller, http:Request request) {
+        http:Response response;
+        json reqPayload;
+
+        string resourcePath = "/airline/asiana";
+        log:printDebug("Received at : " + resourcePath);
+        // Try parsing the JSON payload from the request
+        match request.getJsonPayload() {
+            // Valid JSON payload
+            json payload => reqPayload = payload;
+            // NOT a valid JSON payload
+            any => {
+                response.statusCode = 400;
+                response.setJsonPayload({"Message":"Invalid payload - Not a valid JSON payload"});
+                _ = caller->respond(response);
+                log:printWarn("Invalid payload at : " + resourcePath);
+                done;
+            }
+        }
+
+        string arrivalDate = <string> reqPayload.ArrivalDate but {error => ""};
+        string departureDate = <string> reqPayload.DepartureDate but {error => ""};
+        string rom = <string> reqPayload.From but {error => ""};
+        string to = <string> reqPayload.To but {error => ""};
+        string airline = "Asiana";
+
+        // If payload parsing fails, send a "Bad Request" message as the response
+        if (arrivalDate == null || arrivalDate == null || rom == null || to == null) {
+            response.statusCode = 400;
+            response.setJsonPayload({"Message":"Bad Request - Invalid Payload"});
+            _ = caller->respond(response);
+            log:printWarn("Request with unsufficient info at : " + resourcePath + " : " );
+            done;
+        }
+
+        // Query the database to retrieve flight details
+        json flightDetails = untaint airlineDBService(airline, departureDate, arrivalDate, to, rom);
+        // Response payload
+        log:printDebug("Client response from Asiana : " + flightDetails.toString());
+        response.setJsonPayload(flightDetails);
+        // Send the response to the caller
+        _ = caller->respond(response);
+    }
+
+    // Resource 'flightEmirates', which checks about airline 'Emirates'
+    @http:ResourceConfig {
+        methods:["POST"],
+        path:"/emirates",
+        consumes:["application/json"],
+        produces:["application/json"]
+    }
+    flightEmirates (endpoint caller, http:Request request) {
```

Comment:
You seem to have 3 resources with the same functionality. You can use path params ("/{vendor}") and have one resource. Please check.

Reply:
Adopting this will affect on Jaeger visualizations which conflicts with the intention of this guide.